[darkmoongreatsword] iP

darkmoongreatsword opened the pull request:
#Lewis

>"Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is:

1. download it from [here](https://github.com/darkmoongreatsword/ip).
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x]     Managing tasks
- [ ]     Managing deadlines (coming soon)
- [ ]     Reminders (coming soon)

darkmoongreatsword committed:
-Changed Duke to Lewis

darkmoongreatsword committed:
-Added echo functionality to Lewis

darkmoongreatsword committed:
-Added support for task scheduling.
-Currently 3 types of task: Todos, Events, Deadlines
-Added support for deleting tasks
-Added support for displaying the current tasklist
-Added support to mark tasks as complete or incomplete

Commit:
-Added exception handling for poor inputs for todo, event and deadline for Lewis
-Added semi-automated testing for text UI.

darkmoongreatsword committed:
Update README.md

amoschee commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+import java.util.Arrays;
+import java.util.Scanner;
+public class Lewis {
+    private static boolean exit = false;
```
Consider naming boolean isExit for better clarity

amoschee commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+
+            }
+            case "delete" -> {
+                StringBuilder taskToDelete = new StringBuilder();
```
Variable names for task related stringBuilder does not really sound like a string, maybe you can reconsider naming them taskToDeleteToString?

amoschee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+public abstract class Task {
+    private final String description;
+    protected enum Status {
```
Good naming for the ENUM!

amoschee commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        this.status = Status.NOT_DONE;
+    }
+    protected void updateStatus(Task.Status input) {
+        StringBuilder changelog = new StringBuilder();
```
Similar issue here, this name does not really sound like a string, maybe rename it changeLogToString would make it more intuitive for everyone

amoschee commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    protected int curr = 0, end = 0, len = 0;
```
Naming of these variables seems abit ambiguous, would be better if they reference what curr, end, and len are referring to. (which in this case would be the indexes of tasks)

amoschee reviewed (commented):
Overall LGTM, just need to take note of some naming conventions as well as more JavaDoc comments, good job!

Feng1231 commented on README.md:
Good that you have updated the README.md document!

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+    private static boolean exit = false;
+    protected static TaskList taskList = new TaskList();
+
+    private final static String[] commands = new String[] {"help","mark","unmark","echo","todo","deadline","event","bye","hello","list"};
```
Consider wrapping this for better readability

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+        curr++;
+        String command = str.toString();
+        switch (command) {
+            case "help" -> {System.out.println("Here is the list of valid commands that I know now:");
```
Use Egyptian brackets.

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+                    todoDescript.append(input.charAt(curr));
+                    curr++;
+                }
+                if (todoDescript.isEmpty() || todoDescript.toString().matches("^\\s*$")) throw new LewisException("Hey, a todo can't be empty!");
```
Single line conditional statements should still be wrapped in curly brackets.

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+                while (curr < input.length() && input.charAt(curr) != '/'){
+                    deadlineDescript.append(input.charAt(curr++));
+                }
+                if (deadlineDescript.isEmpty() || deadlineDescript.toString().matches("^\\s*$")) throw new LewisException("Hey, a task can't be empty!");
```
Single line conditional statements should still be wrapped in curly brackets.

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+                    deadlineDescript.append(input.charAt(curr++));
+                }
+                if (deadlineDescript.isEmpty() || deadlineDescript.toString().matches("^\\s*$")) throw new LewisException("Hey, a task can't be empty!");
+                if (curr + 3 >= input.length()) {throw new LewisException("Please type your command in the following form:\n" +
```
Use Egyptian brackets

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+                        System.out.printf("You are about to remove this task:\n%s",taskList.get(index));
+                        taskList.remove(index);
+                    }
+                    else {throw new LewisException(String.format("Hey, please type a valid number between %d and %d.", 1, taskList.len));}
```
if-else statement styling violates the coding standard

Feng1231 commented on src/main/java/Lewis.java:
```diff
@@ -0,0 +1,227 @@
+                    }
+                    else {throw new LewisException(String.format("Hey, please type a valid number between %d and %d.", 1, taskList.len));}
+                }
+                catch (NumberFormatException e) {
```
try-catch block styling violates the coding standard

Feng1231 commented on src/main/java/LewisException.java:
```diff
@@ -0,0 +1,5 @@
+public class LewisException extends Exception{
```
needs a space after Exception

Feng1231 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+public abstract class Task {
+    private final String description;
+    protected enum Status {
+        DONE,NOT_DONE,URGENT
```
needs a space after comma

Feng1231 commented on src/main/java/Lewis.java:
In general, the code looks good! Except a few places with coding standard violations.
1. All conditionals, including the single statement conditionals, should be wrapped in curly braces.
2. To use egyptian brackets whenever it is used
3. Ensure switch and if-else statement styling aligns with the coding standard documentation.

Feng1231 reviewed (commented):
LGTM, not much coding violation, except for some codes in Lewis.java

Commit:
Add save functionality to TaskList.java

Commit:
Add load functionality to TaskList.java

Commit:
Remove redundant code in Lewis.java

Commit:
Merge branch branch-Level-7 to master.

Commit:
Fix NoSuchFileException handling in TaskList.java

Commit:
Add support for storing event times as Java LocalDate objects instead of strings

Commit:
Remove debugging print statements

Commit:
Merge branch 'branch-Level-8' to master

Commit:
Change date formatting for printing dates in Deadline.java and Event.java

Commit:
Merge branch 'branch-Level-8'

Commit:
Add Command.java

Commit:
Add ByeCommand.java

Commit:
Add DeadlineCommand.java

Commit:
Add DeadlineCommand.java

Commit:
Add EchoCommand.java

Commit:
Revert "Add DeadlineCommand.java"

This reverts commit 785ad1d00be3b9fc81ec52510b8c23521e19b808.

Commit:
Add EchoCommand.java

Commit:
Add DeleteCommand.java

Commit:
Add EventCommand.java

Commit:
Add HelloCommand.java

Commit:
Add HelpCommand.java

Commit:
Add ListCommand.java

Commit:
Add MarkUnmarkCommand.java

Commit:
Add TodoCommand.java

Commit:
Add Parser.java

Commit:
Add Storage.java

Commit:
Add Ui.java

Commit:
Clean up code in ./src/main/java

Commit:
Remove Utilities.java in ./src/main/java

Commit:
Revert "Remove Utilities.java in ./src/main/java"

This reverts commit a8971af192966184d7edfda95251b7161fa4a03a.

Commit:
Remove Utilities.java in ./src/main/java

Commit:
Clean up code in ./src/main/java

Commit:
Add test batch file

Commit:
Add gradle files

Commit:
Fix bug in DeadlineCommand.java

Commit:
Package Lewis files into package 'lewis'

Commit:
Merge branch 'branch-A-Packages' into master

Commit:
Add JUnit tests for Deadline and TaskList

Commit:
Merge branch 'branch-A-JUnit' to master

Commit:
Configure build.gradle for jar file

Commit:
Update Javadocs for all files in package 'lewis'

Commit:
Add find command

Commit:
Tweak code in package 'lewis' to comply with coding standard

Commit:
Tweak code in package 'lewis' to comply with coding standard

Commit:
Revert "Configure build.gradle for jar file"

This reverts commit 15cf00fced91ce7317d39898a64b084ce5ad57f8.

Commit:
Revert "Revert "Configure build.gradle for jar file""

This reverts commit cb7070daceb2da50c8123b56dae9367b66833708.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-A-Level-9'

* branch-A-Level-9:
  Add find command

# Conflicts:
#	src/main/java/lewis/Parser.java

Commit:
Add checkstyle.xml and suppression.xml to configs

Commit:
Tweak code to align with style guide

Commit:
Merge branch 'A-CheckStyle'

* A-CheckStyle:
  Tweak code to align with style guide
  Add checkstyle.xml and suppression.xml to configs

Commit:
Implement Stage 2 of GUI

Commit:
Implement Stage 3 of GUI

Commit:
Refactor GUI to use FXML

Commit:
Update GUI to show Lewis responses

Commit:
Merge branch 'branch-Level-10'

Commit:
Implement assertions for code documentation

Commit:
Refactor code to remove redundancy

darkmoongreatsword committed:
Merge pull request #2 from darkmoongreatsword/branch-A-Assertions

Implement assertions for code documentation

Commit:
Refactor code to use streams for conciseness

darkmoongreatsword committed:
Merge pull request #3 from darkmoongreatsword/branch-A-Code-Quality

Refactor code to improve code quality

darkmoongreatsword committed:
Merge pull request #4 from darkmoongreatsword/branch-A-Streams

Refactor code to use streams for conciseness

Commit:
Merge branch 'master' of https://github.com/darkmoongreatsword/ip

Commit:
Implement auto-sort for tasklist

Commit:
Improve auto-sort to include task status

Commit:
Merge branch 'branch-C-Sort'

Commit:
Fix error in Parser

Commit:
Fix title of GUI

Commit:
Update README

Commit:
Update README

Commit:
Merge branch 'master' of https://github.com/darkmoongreatsword/ip

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

Add Windows Key icon under Quick Start instructions, for WIndows users to match the instructions for MacOS users

darkmoongreatsword committed:
Update README.md

Include acknowledgements in README

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md

Update this README to the same one as in ./docs

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Add WindowsKey.png

darkmoongreatsword committed:
Update README.md

Fix missing Windows Key button

darkmoongreatsword committed:
Update README.md

darkmoongreatsword committed:
Update README.md
